pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/IntSumAggregationFunction.java

```diff
@@ -0,0 +1,334 @@
+import org.apache.pinot.core.query.aggregation.groupby.ObjectGroupByResultHolder;
+import org.apache.pinot.segment.spi.AggregationFunctionType;
+import org.apache.pinot.spi.data.FieldSpec.DataType;
+
+
+/**
+ * Specialized INT sum aggregation function that avoids type conversion overhead.
+ * This function is optimized for INT columns and uses native INT arithmetic.
+ *
+ * Performance optimizations:
+ * - Direct INT arithmetic without DOUBLE conversion
+ * - Vectorized operations for better CPU utilization
+ * - Minimal object allocations
+ * - Optimized for the specific case of INT column aggregation
+ * - Proper null handling support using foldNotNull and forEachNotNull
+ */
+public class IntSumAggregationFunction extends NullableSingleInputAggregationFunction<Long, Long> {
+  public static final String FUNCTION_NAME = "intSum";
+
+  public IntSumAggregationFunction(List<ExpressionContext> arguments, boolean nullHandlingEnabled) {
+    super(arguments.get(0), nullHandlingEnabled);
+  }
+
+  @Override
+  public AggregationFunctionType getType() {
+    return AggregationFunctionType.INTSUM;
+  }
+
+  @Override
+  public AggregationResultHolder createAggregationResultHolder() {
+    return new ObjectAggregationResultHolder();
+  }
+
+  @Override
+  public GroupByResultHolder createGroupByResultHolder(int initialCapacity, int maxCapacity) {
+    return new ObjectGroupByResultHolder(initialCapacity, maxCapacity);
+  }
+
+  @Override
+  public void aggregate(int length, AggregationResultHolder aggregationResultHolder,
+      Map<ExpressionContext, BlockValSet> blockValSetMap) {
+    BlockValSet blockValSet = blockValSetMap.get(_expression);
+
+    if (blockValSet.getValueType().getStoredType() != DataType.INT) {
+      throw new IllegalArgumentException("IntSumAggregationFunction only supports INT columns");
+    }
+
+    int[] values = blockValSet.getIntValuesSV();
+
+    // Use foldNotNull with null as initial value - this will return null if no non-null values are processed
+    Long sum = foldNotNull(length, blockValSet, null, (acum, from, to) -> {
+      long innerSum = 0;
+      for (int i = from; i < to; i++) {
+        innerSum += values[i];
+      }
+      return acum == null ? innerSum : acum + innerSum;
+    });
+
+    // Update the result holder
+    ObjectAggregationResultHolder objectHolder = (ObjectAggregationResultHolder) aggregationResultHolder;
+    Object existingResult = objectHolder.getResult();
+    long existingSum = 0L;
+    if (existingResult != null) {
+      if (existingResult instanceof Long) {
+        existingSum = (Long) existingResult;
+      } else if (existingResult instanceof Double) {
+        existingSum = ((Double) existingResult).longValue();
+      } else if (existingResult instanceof Number) {
+        existingSum = ((Number) existingResult).longValue();
+      } else {
+        // If it's some other type, try to parse it as a number
```

Comment:
Why are these checks required? The aggregation result holder should always have a `Long` right?